graylog2-web-interface/src/components/configurations/UrlWhiteListForm.jsx

```diff
@@ -0,0 +1,204 @@
+    _validate(event.target.name, idx, type, FormUtils.getValueFromInput(event.target));
+  };
+
+  const _onUpdateType = (idx: number, type: string) => {
+    const stateUpdate = ObjectUtils.clone(state);
+    _validate('value', idx, type, stateUpdate.entries[idx].value);
+  };
+  const _summary = () => {
+    return (state.entries.map((url, idx) => {
+      return (
+      // eslint-disable-next-line react/no-array-index-key
+        <tr key={url.id}>
+          <td>{idx + 1}</td>
+          <td>
+            <Input type="text"
+                   id={`title-input${idx}`}
+                   ref={(elem) => { inputs[`title${idx}`] = elem; }}
+                   help={validationState.errors[idx] && validationState.errors[idx].title && !validationState.errors[idx].title.valid ? 'required' : null}
+                   name="title"
+                   bsStyle={validationState.errors[idx] && validationState.errors[idx].title && !validationState.errors[idx].title.valid ? 'error' : null}
+                   onChange={event => _onInputChange(event, idx, url.type)}
+                   defaultValue={url.title}
+                   required />
+          </td>
+          <td>
+            <Input type="text"
+                   id={`value-input${idx}`}
+                   ref={(elem) => { inputs[`value${idx}`] = elem; }}
+                   help={validationState.errors[idx] && validationState.errors[idx].value && !validationState.errors[idx].value.valid ? 'Not valid' : null}
+                   name="value"
+                   bsStyle={validationState.errors[idx] && validationState.errors[idx].value && !validationState.errors[idx].value.valid ? 'error' : null}
+                   onChange={event => _onInputChange(event, idx, url.type)}
+                   defaultValue={url.value}
+                   required />
+          </td>
+          <td>
+            <Input id={`url-input-type-${idx}`}
+                   required
+                   autoFocus>
+              <Select placeholder="Select Cache Type"
+                      clearable={false}
+                      options={options}
+                      matchProp="label"
+                      onChange={option => _onUpdateType(idx, option)}
+                      value={url.type} />
+            </Input>
+          </td>
+          <td>
+            <span className="">
+              <Icon name="fa-trash" style={{ cursor: 'pointer' }} onClick={event => _onRemove(event, idx)} />
+            </span>
+          </td>
+        </tr>
+      );
+    }));
+  };
+
+  useEffect(() => {
+    const valid = _isFormValid();
+    update(state, valid);
+  }, [state]);
+
+  return (
+    <>
+      <Input type="checkbox"
+             id="whitelist-disabled"
+             label="Disabled"
+             checked={state.disabled}
+             onChange={() => setState({ ...state, disabled: !state.disabled })}
+             help="Disable this white list." />
+      <Table striped bordered className="top-margin">
```

Comment:
I find myself scrolling horizontally inside the inputs to see the full titles and URLs and I think that will be frustrating for users. I'm not sure what the time constrains on this will allow us but I think we should explore other options to place the inputs, giving them more space. What do you think?

Reply:
Yes i though about it, and already considered having an edit page instead of modal.
But for the sake of consistancy with other config i think modal is better that's why i maxed out the modal size to "lg". We can this also and see other options.

Reply:
Yes, using the modal limits us quite a bit, and adding inputs vertically will just make people scroll a lot to find what they are looking for. Let's leave it like this for now, specially when we need to backport it, and think about it for 3.2 if we have some time later on.